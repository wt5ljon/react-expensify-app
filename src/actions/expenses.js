import uuid from 'uuid';
import database from '../firebase/firebase';

// ADD_EXPENSE
export const addExpense = (expense) => ({
  type: 'ADD_EXPENSE',
  expense
});

export const startAddExpense = (expenseData = {}) => {
  return (dispatch) => {
    const {
      description = '',
      note = '',
      amount = 0,
      createdAt = 0
    } = expenseData;
    const expense = { description, note, amount, createdAt };
    return database.ref('expenses').push(expense).then((snapshot) => {
      dispatch(addExpense({
        id: snapshot.key,
        ...expense
      }));
    })
  };
};

// REMOVE_EXPENSE
export const removeExpense = ({ id }) => ({
  type: 'REMOVE_EXPENSE',
  id
});

// EDIT_EXPENSE
export const editExpense = (id, update) => ({
  type: 'EDIT_EXPENSE',
  id,
  update
});
