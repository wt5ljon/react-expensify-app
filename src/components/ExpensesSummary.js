import React from 'react';
import { connect } from 'react-redux';
import SelectExpensesTotal from '../selectors/expenses-total';
import SelectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary  = ({ expenseTotal, expenseCount }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const total = numeral(expenseTotal / 100).format('$0, 0.00');
  return (
    <div>
      <h1>Viewing {expenseCount} {expenseWord} totaling {total}</h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = SelectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expenseTotal: SelectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);