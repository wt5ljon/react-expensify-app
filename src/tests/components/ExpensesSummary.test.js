import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary correctly for no expenses', () => {
  const wrapper = shallow(<ExpensesSummary 
    expenseTotal={0}
    expenseCount={0}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly for one expense', () => {
  const wrapper = shallow(<ExpensesSummary
    expenseTotal={4500}
    expenseCount={1}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly for multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary
    expenseTotal={109500}
    expenseCount={2}
  />);
  expect(wrapper).toMatchSnapshot();
});