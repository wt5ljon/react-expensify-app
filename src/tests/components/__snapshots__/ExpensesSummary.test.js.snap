// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly for multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totaling 
    $1,095.00
  </h1>
</div>
`;

exports[`should render ExpensesSummary correctly for no expenses 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expenses
     totaling 
    $0.00
  </h1>
</div>
`;

exports[`should render ExpensesSummary correctly for one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $45.00
  </h1>
</div>
`;
